pymc_bart/split_rules.py

```diff
@@ -0,0 +1,71 @@
+#   Copyright 2022 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from numba import njit
+import numpy as np
+
+class ContinuousSplitRule:
+    """
+    Standard continuous split rule: pick a pivot value and split depending on if variable is smaller or greater than the value picked.
+    """
+
+    @staticmethod
+    def get_split_value(available_splitting_values):
+        split_value = None
+        if available_splitting_values.size > 1:
+            idx_selected_splitting_values = int(np.random.random() * len(available_splitting_values))
+            split_value = available_splitting_values[idx_selected_splitting_values]
+        return split_value
+
+    @staticmethod
+    @njit
+    def divide(available_splitting_values, split_value):
+        return available_splitting_values <= split_value
+
+class OneHotSplitRule:
+    """Choose a single categorical value and branch on if the variable is that value or not"""
+
+    @staticmethod
+    def get_split_value(available_splitting_values):
+        split_value = None
+        if available_splitting_values.size > 1 and not np.all(available_splitting_values==available_splitting_values[0]):
+            idx_selected_splitting_values = int(np.random.random() * len(available_splitting_values))
+            split_value = available_splitting_values[idx_selected_splitting_values]
+        return split_value
+
+    @staticmethod
+    @njit
+    def divide(available_splitting_values, split_value):
+        return available_splitting_values == split_value
+
+class SubsetSplitRule:
+    """
+    Choose a random subset of the categorical values and branch on if the value is within the chosen set.
+    This is the approach taken in flexBART paper.
```

Comment:
Please add a reference to the paper including name of the article, authors and DOI.

Reply:
It's this paper: https://arxiv.org/abs/2211.04459
Do you have an example of citation formating somewhere so I can align the style?